tests/jerry/es2015/regression-test-issue-2757.js

```diff
@@ -0,0 +1,23 @@
+// Copyright JS Foundation and other contributors, http://js.foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+var v0 = Function ('return this;')( );
+var v1 = (new Int8Array (149)).subarray (78);
+
+try {
+  var $ = $(v1.set (v1));
+  assert (false);
+} catch (e) {
+  assert (e instanceof TypeError);
```

Comment:
With this we are testing that the `$(..)` call triggers a TypeError. The test case should be a minimal set of operations related to the modification.